index.js

```diff
@@ -1,48 +1,53 @@
-const { Client, Attachment } = require('discord.js');
-const https = require('https');
-const client = new Client();
-const prefix = 'hack!';
+const Discord = require('discord.js');
+const client = new Discord.Client();
+const config = require("./config.json");
+const prefix = config.prefix;
+client.config = config;
+
+const Enmap = require("enmap");
+const fs = require("fs");
+client.commands = new Enmap();
+fs.readdir("./commands/", (err, files) => {
+  if(err)
+    return console.error(err);
+
+  files.forEach(file => {
+    if (!file.endsWith(".js"))
+      return;
+
+    let props = require(`./commands/${file}`);
+    let commandName = file.split(".")[0];
+
+    console.log(`Attempting to load command ${commandName}`);
+    client.commands.set(commandName, props);
+  });
+});
 
 client.on('ready', () => console.log('Bot has logged in!'));
 
 client.on('message', async (msg) => {
   if(msg.author.bot || !msg.content.startsWith(prefix)) return;
   let args = msg.content.split(' ');
   const command = args.shift().slice(prefix.length);
+  
   //Add your commands here. Good command handlers are overrated :POGGERS:
+  const cmd = client.commands.get(command);
+  if(!cmd)
+    return;
+
+  if(cmd.adminOnly)
+    if(!msg.member.hasPermission(Permissions.ADMINISTRATOR))
+      msg.reply(`You have to be admin to use the command \`${cmd.name}\``);
```

Comment:
you realize this isn't exiting the message handler so it will still run the commands, right?